### PR TITLE
NEAREST_CACHE preprend cache list when specified

### PR DESCRIPTION
**Pull request by joereuss12**
To specify the local cache, users need to set the NEAREST_CACHE env variable. In the old functionality, setting this means we do not look for any other caches and just try to use this "preferred cache". Now, if the specified NEAREST_CACHE has a '+' character at the end of it, it will now append the "normal" list of caches behind the preferred cache in the list of caches to try.

### CODE REVIEW

**Comment by bbockelm** [comment]
@joereuss12 -
- Can you link this to the corresponding issue?  At least from the mobile UI it appears it's not linked.
- I think the original idea was to treat NEAREST_CACHE as a list and to make the + a special list entry.  Can we still do that approach?  That'll allow multiple site-local caches.

**Comment by joereuss12** [comment]
> @joereuss12 -
> 
>     * Can you link this to the corresponding issue?  At least from the mobile UI it appears it's not linked.
> 

It says it is linked under development, but here: #1155

**Comment by joereuss12** [comment]
>     * I think the original idea was to treat NEAREST_CACHE as a list and to make the + a special list entry.  Can we still do that approach?  That'll allow multiple site-local caches.

Ah, I see what you mean. Yes! I can work that into the implementation!

File: cmd/object_copy.go [hunk 1]
```diff
@@ -181,12 +181,15 @@ func copyMain(cmd *cobra.Command, args []string) {
 	}
 	caches := make([]*url.URL, 0, 1)
 	if preferredCache != "" {
-		if preferredCacheURL, err := url.Parse(preferredCache); err != nil {
-			log.Errorf("Unable to parse preferred cache (%s) as URL: %s", preferredCache, err.Error())
-			os.Exit(1)
-		} else {
-			caches = append(caches, preferredCacheURL)
-			log.Debugln("Preferred cache for transfer:", preferredCacheURL)
+		cacheList := strings.Split(preferredCache, ",")
```
Comment (haoming29) [inline cmd/object_copy.go:184]: Can we pull this logic into a function? I saw this was repeated a couple of times in different places so I'd be nice if the logic can be shared.

File: cmd/object_get.go [hunk 2]
```diff
@@ -99,12 +100,15 @@ func getMain(cmd *cobra.Command, args []string) {
 	}
 	caches := make([]*url.URL, 0, 1)
 	if preferredCache != "" {
-		if preferredCacheURL, err := url.Parse(preferredCache); err != nil {
-			log.Errorf("Unable to parse preferred cache (%s) as URL: %s", preferredCache, err.Error())
-			os.Exit(1)
-		} else {
-			caches = append(caches, preferredCacheURL)
-			log.Debugln("Preferred cache for transfer:", preferredCacheURL)
+		cacheList := strings.Split(preferredCache, ",")
```
Comment (haoming29) [inline cmd/object_get.go:103]: same here

File: cmd/plugin.go [hunk 3]
```diff
@@ -384,14 +384,17 @@ func runPluginWorker(ctx context.Context, upload bool, workChan <-chan PluginTra
 	}()
 
 	caches := make([]*url.URL, 0, 1)
+	var nearestCacheURL *url.URL
```
Comment (haoming29) [inline cmd/plugin.go:387]: here as well

**Review by haoming29 — CHANGES_REQUESTED** [review]
One relatively minor change requested otherwise LGTM

**Comment by haoming29** [comment]
You may also want to merge/rebase against main to let the macOS test pass

**Review by haoming29 — APPROVED** [review]
Revised code LGTM

**Pull request merged**